components/forms/TextPage/TextPage.tsx

```diff
@@ -15,12 +14,19 @@ interface TextPageProps {
   htmlEmail: string | undefined;
 }
 
-const getPageContent = (t: TFunction, pageText: string, urlQuery: string | null) => {
+interface PageContextProps {
+  pageText: string;
+  urlQuery: string | null;
+}
+
+const PageContent = ({ pageText, urlQuery }: PageContextProps) => {
   // Check if there's a custom text for the end page specified in the form's JSON config
   if (pageText && pageText !== undefined) {
     return <RichText className="confirmation">{pageText}</RichText>;
   }
 
+  const { t } = useTranslation("confirmation");
```

Comment:
If the components are on the same file. Does it not make sense to just pass the t function from the parent TextPage to the child PageContent. Seems to be a duplication otherwise

Reply:
My thought process was the separation of scope and self containment of components.  Efficiency doesn't really come into play in the sense that each useTranslation hook is coupled with Suspense.  The hook simply accesses the instance that already exists on the Page context.  By including it on each functional component that requires translation it simplifies the props and TypeScript typing needed for each component declaration.
Just my 2 cents.

Reply:
Think it's a fairly minor detail then. If the useTranslation hook doesn't add any time to the rendering or causes a re-render it can go either way. But in general this is something to think about when deciding whether to pass down or use a hook. I find often times devs use one or the other without thinking about renders or performance ( not saying thats you :p  just mean in general )